clients/src/main/java/org/apache/kafka/common/Cluster.java

```diff
@@ -28,13 +28,14 @@
  */
 public final class Cluster {
 
+    private final boolean isBootstrapConfigured;
     private final List<Node> nodes;
     private final Set<String> unauthorizedTopics;
     private final Map<TopicPartition, PartitionInfo> partitionsByTopicPartition;
     private final Map<String, List<PartitionInfo>> partitionsByTopic;
     private final Map<String, List<PartitionInfo>> availablePartitionsByTopic;
-    private final Map<Integer, List<PartitionInfo>> partitionsByNode;
-    private final Map<Integer, Node> nodesById;
+    private final Map<String, List<PartitionInfo>> partitionsByNodeId;
+    private final Map<String, Node> nodesById;
```

Comment:
I'm actually tempted to call parseInt() within NetworkClient and revert the changes here in order to reduce the scope of this patch.